command.go

```diff
@@ -104,7 +104,18 @@ func (c *Command) Call(req *Request, re ResponseEmitter, env Environment) {
 		}
 	}
 
-	cmd.Run(req, re, env)
+	err = cmd.Run(req, re, env)
+	if err != nil {
+		if cmderr, ok := err.(cmdkit.Error); ok {
+			err = re.Emit(cmderr)
+			if err != nil {
+				log.Error("error %q emitting cmd error %q on request %#v", err, cmderr, req)
```

Comment:
I'm worried we'll hit this every time a client disconnects. What do you think?

Reply:
Hmm, a client disconnect shouldn't trigger an error in Run in the first place, so this entire branch shouldn't trigger - or am I missing something?

Reply:
Client disconnects -> context gets canceled -> Run returns the context cancellation error.

Reply:
Hmm, but usually the server will write their response and return. What you write only happens if the server doesn't expect the client to exit, which is a rightful error IMHO.
If the Run function sends data and expects that the client might have closed the connection by now, they should not return context cancellation errors.

Reply:
> If the Run function sends data and expects that the client might have closed the connection by now, they should not return context cancellation errors.

Telling whether or not the client has gone away can sometimes be a bit tricky and we definitely don't want to return "success" to the user if we *think* the client has disconnected but it hasn't.

Currently, we *always* return an error if we abort the request due to a client disconnect.

Reply:
Okay, so nothing changes then, right?

Reply:
I guess in *this* situation, we're unlikely to return a cmdkit Error so we shouldn't hit this log message.

Although, I still don't think that this is a true error case (for the daemon) as the client could have disconnected after run completed but before we try to send the value back. Really, I think that's the only possible way to hit this log.